src/main/java/sign_up/interface_adapters/SignUpViewModel.java

```diff
@@ -0,0 +1,35 @@
+package sign_up.interface_adapters;
+
+import sign_up.use_case.*;
+
+public class SignUpViewModel implements SignUpOutputBoundary {
```

Comment:
I think you might be able to redesign the UI/interface adapter interactions by implementing the observer pattern by using stuff from the `java.beans` package instead (see [design pattern repo](https://github.com/CSC207-UofT/design-pattern-samples/tree/main/src/observer/withObserverJava9AndLater)).

Reply:
Okay, I'll look into this. Thanks!